examples/settings_list.rs

```diff
@@ -238,13 +239,20 @@ impl Component for SettingsListModel {
         }
     }
 
-    fn command(message: Self::Command) -> CommandFuture<Self::CommandOutput> {
+    fn command(message: Self::Command, mut shutdown: ShutdownReceiver) -> CommandFuture<Self::CommandOutput> {
         Box::pin(async move {
-            match message {
-                SettingsListCommand::Reload => {
-                    tokio::time::sleep(std::time::Duration::from_secs(2)).await;
-                    Some(SettingsListCmdOutput::Reload)
+            let task = async move {
+                match message {
+                    SettingsListCommand::Reload => {
+                        tokio::time::sleep(std::time::Duration::from_secs(2)).await;
+                        Some(SettingsListCmdOutput::Reload)
+                    }
                 }
+            };
+
+            futures::select! {
+                _ = shutdown.recv().fuse() => None,
+                response = task.fuse() => response,
```

Comment:
Can we make this easier? I think most components would just like to drop the future when the shutdown signal is received.